app/components/post_list/post/footer/footer.tsx

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo} from 'react';
+import {TouchableOpacity, View} from 'react-native';
+
+import {updateThreadFollowing} from '@actions/remote/thread';
+import CompassIcon from '@components/compass_icon';
+import FormattedText from '@components/formatted_text';
+import UserAvatarsStack from '@components/user_avatars_stack';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+import type ThreadModel from '@typings/database/models/servers/thread';
+import type UserModel from '@typings/database/models/servers/user';
+
+type Props = {
+    participants: UserModel[];
+    teamId?: string;
+    testID: string;
+    thread: ThreadModel;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    const followingButtonContainerBase = {
+        justifyContent: 'center',
+        height: 32,
+        paddingHorizontal: 12,
+    };
+
+    return {
+        container: {
+            flexDirection: 'row',
+            alignItems: 'center',
+            minHeight: 40,
+        },
+        avatarsContainer: {
+            marginRight: 12,
+            paddingVertical: 8,
+        },
+        replyIconContainer: {
+            top: -1,
+            marginRight: 5,
+        },
+        replies: {
+            alignSelf: 'center',
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            marginRight: 12,
+            ...typography('Heading', 75),
+        },
+        notFollowingButtonContainer: {
+            ...followingButtonContainerBase,
+            paddingLeft: 0,
+        },
+        notFollowing: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Heading', 75),
+        },
+        followingButtonContainer: {
+            ...followingButtonContainerBase,
+            backgroundColor: changeOpacity(theme.buttonBg, 0.08),
+            borderRadius: 4,
+        },
+        following: {
+            color: theme.buttonBg,
+            ...typography('Heading', 75),
+        },
+        followSeparator: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.16),
+            height: 16,
+            marginRight: 12,
+            width: 1,
+        },
+    };
+});
+
+const Footer = ({participants, teamId, testID, thread}: Props) => {
+    const serverUrl = useServerUrl();
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+    const toggleFollow = useCallback(preventDoubleTap(() => {
+        if (!teamId) {
```

Comment:
Will DM/GM's have this value assigned as it is from thread_in_team?

Reply:
OK following the code we get the teamId based on the channel the post belongs to, for DM/GM's this is an empty string so this check won;t allow me to follow/unfollow, is that expected?

Reply:
Sorry, I was still fixing the issue. That's why I haven't replied yet. Yes, DM/GMs have an issue with this approach and I am addressing it 🙂 Thank you!

Reply:
`thread_in_team` will have only the threads from `GlobalThreads`. So, we won't get the teamId from that table either instead I am sending currentTeamId to support both the following and not-following threads.